Sources/ProjectDescription/TargetDependency.swift

```diff
@@ -53,7 +53,8 @@ public enum TargetDependency: Codable, Hashable {
     /// - Parameters:
     ///   - product: The name of the output product. ${PRODUCT_NAME} inside Xcode.
     ///              e.g. RxSwift
-    case package(product: String)
+    ///   - isPlugin: Must be explicitly set to `true` if this is a plugin to generate the Xcode project correctly.
+    case package(product: String, isPlugin: Bool = false)
```

Comment:
Not super familiar with SPM plugins. Is a package either a plugin or a normal package? Or it can be both?

Reply:
a package contains one or more products and a product can be a either an executable, a library, or a plugin.

Reply:
Yes, then it's a bit strange to have an `isPlugin` boolean as the package itself is not a plugin

Reply:
no, because this case does not refer to a package, but to a product (within a package).
the enum case name is a bit confusing, I think.

Reply:
I just discovered `PackageInfo.ProductType` in `TuistSupport` target and it looks like I should be somehow able to access that to find out if a certain product is a plugin or not, so that the user does not have to specify it and I could get rid of this `isPlugin` flag entirely.

I just need to find connect this somehow and am not quite sure if this is possible and how. Any ideas/suggestions?

Reply:
`PackageInfo` is for the newer Dependenceis.swift dependencies management, you don't have access to that whe using the legacy `.package` dependency

Reply:
OK, so ideally you would not have to change anything and Tuist would just know if a certain SPM package product is a plugin or not, but since this does not seem to be possible I think the next best way to solve this is to have to user tell via an optional `isPlugin` flag.

Should it become possible for Tuist will figure out the product type at some point, it would not be a breaking change. Setting `isPlugin` would just become deprecated and be optional.

Reply:
No strong opinions here though. I'm happy to adapt this to whatever the core team prefers. I would just like to get this merged soon, if possible :)

Reply:
Note that there are two ways of adding packages to Tuist project:
- `Dependencies.swift`, which @danyf90 is referring to.
- Xcode's approach, which you are touching on this PR.

I doubt we would be able to support plugins or macros with the former because it maps packages to Xcode targets, which don't support plugins or macros. The latter option delegates package resolution to Xcode, so at generation time we don't have information to determine if we should set or not a flag in the generated `pbxproj` file.

Reply:
OT, but: does that mean that projects using `Dependencies.swift` will not be able to make use of both SPM plugins and also macros? this could be big issue, no?

Reply:
Exactly. We are aware of the issue and we are putting some thought into how to overcome the issue. Apple is strongly coupling new build-time features with Swift Packages, and that conflicts with our `Dependencies.swift` approach that uses Xcode projects at its core. A solution to this problem would require figuring out a map from Packages to Xcode Projects' domain.